zebra-chain/src/block/header.rs

```diff
@@ -35,10 +34,19 @@ pub struct BlockHeader {
     /// header.
     pub merkle_root_hash: MerkleTreeRootHash,
 
-    /// [Sapling onward] The root LEBS2OSP256(rt) of the Sapling note
+    /// [Pre-Sapling] Reserved. All zeroes.
+    /// [Sapling and Blossom] The root LEBS2OSP256(rt) of the Sapling note
     /// commitment tree corresponding to the final Sapling treestate of
     /// this block.
-    pub final_sapling_root_hash: SaplingNoteTreeRootHash,
+    /// [Heartwood onward] The root of a Merkle Mountain Range tree, which
+    /// commits to various features of the chain's history, including the
+    /// Sapling commitment tree. This commitment supports the FlyClient
+    /// protocol. See ZIP-221 for details.
+    // TODO:
+    //   - replace with an unspecified HistoryRootHash type?
```

Comment:
Yeah I would prefer to have a refined type here, although it's annoying that this sort of change isn't versioned, rather it's overloading the meaning of the same 32 bytes (it could be a Sapling root, or a Heartwood MMR: which is it? we'll find out!) 😒

Reply:
Even worse, we don't know what it is, until we've parsed the transactions and got the coinbase height.

And in Zebra, the network upgrade coinbase heights are in zebra-consensus, but the parsing is done in zebra-chain.

Reply:
I'm sorry. FWIW the rationale for not doing a version bump is at https://zips.z.cash/zip-0221#header-format-change

Reply:
@teor2345 we could add an opaque refinement type for the parsing step in `zebra-chain` which then has methods to parse it into the correct refinement type when given the requisite context as args.

Reply:
That's the plan, see https://github.com/ZcashFoundation/zebra/pull/768#issue-457081416.

As soon as we are happy with this PR, I'll make those changes in #768.